Plugins/ServerPlugin.cpp

```diff
@@ -148,6 +154,7 @@ void CompilerClient::CompileBuffer(Game* game, CompileMode mode, std::string nam
 
   auto worker = dynamic_cast<AsyncReadWorker<CompileReply>*>(callData);
   worker->stream = stub->PrepareAsyncCompileBuffer(&worker->context, request, &cq);
+  callData->start();

```

Comment:
Well that's not how Google does it?
https://github.com/grpc/grpc/blob/694f491e06b3503952ca5053c98493cf48af43c1/examples/cpp/helloworld/greeter_async_client2.cc#L54

Anyway, that would break everything? Then worker would destruct as soon as this returns, which it's going to because it's not blocking and async.

Reply:
The worker would be a thin wrapper around that pointer which hides that cast and calls start() on destruct.

Reply:
Ok that is an interesting idea but it's a little confusing because callData literally is a worker. CallData is the abstract base class that both async workers inherit from. It would be interesting to combine this together with ScheduleTask if we could.